### PR TITLE
Use Correct Dependency Type for Gatsby-Source-Filesystem

### PR DESCRIPTION
**Pull request by NWRichmond**
**Story:** [ENG-233](https://nacelle.atlassian.net/browse/ENG-233) 

> [gatsby-source-nacelle] don't include gatsby-source-filesystem by default

### Type of Change

- [ ] Bug fix
- [ ] Non-breaking feature
- [ ] Breaking feature
- [x] Chore (docs, refactor, etc)

### What is being changed and why?

We want devs using `gatsby-source-nacelle` to be able to opt in to using Gatsby Image. Currently, devs installing `gatsby-source-nacelle` will unwittingly be opting in to using Gatsby Image unless they install via `@nacelle/gatsby-source-nacelle` with the [`--no-optional`](https://docs.npmjs.com/cli/v6/commands/npm-install) flag.

This PR removes `gatsby-source-filesystem` from the [`optionalDependencies`](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#optionaldependencies) and adds it to the [`peerDependencies`](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#peerdependencies). This way, `gatsby-source-filesystem` is not installed via `npm i @nacelle/gatsby-source-nacelle` and requires explicit installation as described in `packages/gatsby-source-nacelle/README.md`.

#### BONUS

Slight refactor of `packages/gatsby-source-nacelle/src/utils/createRemoteImageFileNode.js` which accomplishes the following:

- Avoids nested `try { ... } catch(err) { ... }`
- Shifts the responsibility to check for the presence of `gatsby-source-filesytem` to a single file (`packages/gatsby-source-nacelle/gatsby-node.js`) rather than multiple files
- Improves image download performance by parallelizing `createRemoteFileNode` calls in a `Promise.all()`

### How to Test

Unfortunately, running `npm run bootstrap` from the monorepo root will still install `gatsby-source-filesystem` to `examples/gatsby` as per the published `@nacelle/gatsby-source-nacelle@2.9.0`'s `optionalDependencies` array. The local `packages/gatsby-source-nacelle/package.json` doesn't seem to be respected, and I'm not sure why.

In other words, I think this will work as expected once published, but I'm having difficulty getting it to work locally due to Lerna quirks.

### CODE REVIEW

**Review by johnstonbl01 — APPROVED** [review]
Looks good to me!

When running `npm run bootstrap`, keep in mind that Lerna will hoist dependencies, so you would see `gatsby-source-filesystem` in `node_modules` because it's a direct dependency of the `examples/gatsby` project. LMK if you want to chat about this more!

**Comment by NWRichmond** [comment]
> LMK if you want to chat about this more!

@johnstonbl01 that would be great! I'm still flummoxed by how `gatsby-source-filesystem` is being installed after doing the following:

```
cd examples/gatsby
npm rm gatsby-source-filesystem
rm -rf node_modules
cd ../../
npm run bootstrap
```

When I do that, I can tell that it's still being installed because right below [this line](https://github.com/getnacelle/nacelle-react/blob/2507fb297bca13ae7a4be96c393bf0390516fb7b/packages/gatsby-source-nacelle/gatsby-node.js#L97) I can add:
```
console.log(`VERSION: ${require('gatsby-source-filesystem/package.json').version}`);
```
...and then in `examples/gatsby` when I run `npm run dev` it prints `VERSION: 2.8.1`.

For reference, `npm ls gatsby-source-filesystem` from `examples/gatsby` reveals:
Screen Shot 2021-01-04 at 3 57 56 PM

**Comment by johnstonbl01** [comment]
I think you may just be running into an order of operations problem with Lerna.

Try this:

```bash
# /nacelle-react/examples/gatsby
npm uninstall gatsby-source-filesystem -S
cd ../..
npx lerna clean
rm -rf node_modules && npm i
npm run bootstrap
find . -name package.json  -exec grep -l gatsby-source-filesystem /dev/null {} \;
```

The result of `find` for me after doing the above is just the peerdependency in the `gatsby-source-nacelle` `package.json`.

**Comment by NWRichmond** [comment]
Thank you, @johnstonbl01! `lerna clean` was the missing step. I've added a `clean` command to the project root.
